src/Microsoft.PowerShell.PSReadLine/Render.cs

```diff
@@ -383,7 +383,7 @@ private void GetTokenColors(Token token, out ConsoleColor foregroundColor, out C
                 return;
             }
 
-            if ((token.TokenFlags & TokenFlags.CommandName) != 0)
+            if ((token.TokenFlags & TokenFlags.CommandName) != 0 && token.Kind != TokenKind.DynamicKeyword)
```

Comment:
I noticed that in code that pre-exists, the `Command`-bodied keywords are parsed as commands, and [then flagged that way](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/parser/Parser.cs#L5511) when they are parsed by the command rule. It also specifically [unflags commands as keywords](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/parser/Parser.cs#L5540).

I added naive exceptions for `DynamicKeyword`s to make the highlighting consistent because `CommandRule` seemed to be already aware of `DynamicKeyword`s and I assumed `GetCommandArgument` was similar.

But, I certainly agree that it would be simpler to have `TokenFlag.CommandName` set, and can make that change in `GetCommandArgument` to not set the flag if the token is a `DynamicKeyword`.